### PR TITLE
posts and new tab titles

### PR DESCRIPTION
**Pull request by ethn**
This adds posts as described here: https://www.pivotaltracker.com/story/show/156223859

It also standardizes tabs following the proposal I put out in slack:
> ALL TABS: 2 lines:
  top: count (optional) & icon (required)
  bottom: title (required) (edited)
To do this I had to pick some new icons, all of which are now represented alongside the old in a single icon map. I expect @srivig will want to review and improve:
1. the post pages 
2. the tab design
3.  the icons

I want to add some tests using philipp's new test API

### CODE REVIEW

File: mod/wikirate/set/type/wikirate_topic/structure.rb
Comment (ethn) [inline mod/wikirate/set/type/wikirate_topic/structure.rb:10]: tab lists are now actually just lists.

File: mod/wikirate/set/all/wikirate/icons.rb
Comment (ethn) [inline mod/wikirate/set/all/wikirate/icons.rb]: @srivig, this is the icon map.

File: mod/posts/set/all/post.rb
Comment (ethn) [inline mod/posts/set/all/post.rb:8]: this post tab is here because it's the same for all types.  We might be able to repeat something like this pattern for other types.

File: mod/layout/template/abstract/listing/listing.haml
Comment (srivig) [inline mod/layout/template/abstract/listing/listing.haml]: This change would break the responsiveness.

**Review by xithan — CHANGES_REQUESTED** [review]
I like the tab api improvements but I'm waiting for a test coverage greater than 0 before I approve this.

**Comment by ethn** [comment]
over 50 (generic) specs added.

**Review by xithan — APPROVED** [review]
👍

**Pull request merged**
